docs/control_tuple_support.md

```diff
@@ -0,0 +1,207 @@
+
+Example: Below code snippet illustrates the use of `processControl` method of `ControlAwareDefaultInputPort` to receive / handle user defined control tuples.
+
+```
+public final transient ControlAwareDefaultInputPort<Double> input =
+    new ControlAwareDefaultInputPort<Double>()
+{
+  // Process a data tuple
+  @Override
+  public void process(Double tuple)
+  {
+    output.emit(tuple);
+  }
+
+  // Process a control tuple
+  @Override
+  public boolean processControl(ControlTuple userControlTuple)
+  {
+    // process control tuple here
+    return false;
+    // indicates whether or not the engine
+    // should propagate the tuple automatically to downstream operators
+    // Discussed in later sections
+  }
+};
+
+```
+
+#### Return value of `processControl`
+
+Once the control tuple is processed in the `processControl` method, a return value is expected by the engine. This return value indicates whether or not the operator wishes to handle the propagation of the control tuple or let the engine proceed with the default auto-propagation of the control tuple.
+Following are the semantics:
+1. true - Operator would handle propagation explicitly
+2. false - Operator would not handle propagation. Engine can automatically forward.
+
+See [Propagation of Control Tuples](#propagation-of-control-tuples) for more details
+
+### Serialization requirements
+A control tuple generated by some operator of the application needs to traverse the same path as that traversed by other data tuples transmitted by the application. For this reason, similar to the other data tuples, the control tuple needs to be Kryo serializable since the default serializer used by the platform is Kryo.
+
+
+## Propagation of Control Tuples
+
+A control tuple emitted by an operator can be propagated downstream automatically. This is in line with the automatic propagation of other pre-defined control tuples in the engine. However, some use cases require that the control tuple need not be propagated further in the DAG. We support this behavior for user defined control tuples.
+The `processControl` method of the `ControlAwareDefaultInputPort` retures a boolean return value.
+
+```
+@Override
+public boolean processControl(ControlTuple userControlTuple)
+{
+  // process userControlTuple here
+  // return true if operator wants to propagate explicitly or block propagation
+  // return false if operator wants engine to propagate automatically
+}
+```
+
+### Non - *Control Aware* ports
+For operators, which do not have *Control Aware* ports, just forward the control tuples to the downstream operators implicitly. The application writer / user does not have to worry about handling a Control tuple which is generated upstream. Only operators with *Control Aware* ports would be delivered the control tuple via the `processControl` method.
```

Comment:
Suggest replacing first sentence with:
"For operators without *Control Aware* ports, the platform will forward the control tuples to the downstream operators automatically."